src/main/java/com/artipie/asto/memory/InMemoryStorage.java

```diff
@@ -0,0 +1,103 @@
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.asto.memory;
+
+import com.artipie.asto.Key;
+import com.artipie.asto.Remaining;
+import com.artipie.asto.Storage;
+import com.artipie.asto.Transaction;
+import com.google.common.primitives.Bytes;
+import io.reactivex.Flowable;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Flow;
+import org.reactivestreams.FlowAdapters;
+
+/**
+ * Simple implementation of Storage that holds all data in memory.
+ *
+ * @since 0.1
+ */
+public final class InMemoryStorage implements Storage {
+
+    /**
+     * Values stored by key strings.
+     */
+    private final Map<String, byte[]> data = new HashMap<>();
+
+    @Override
+    public CompletableFuture<Boolean> exists(final Key key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Collection<Key>> list(final String prefix) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Void> save(final Key key, final Flow.Publisher<ByteBuffer> content) {
+        return CompletableFuture.runAsync(
+            () -> {
+                final byte[] bytes = Flowable.fromPublisher(FlowAdapters.toPublisher(content))
+                    .map(buf -> new Remaining(buf).bytes())
+                    .toList()
+                    .blockingGet()
+                    .stream()
+                    .reduce((b1, b2) -> Bytes.concat(b1, b2))
+                    .orElse(new byte[0]);
+                this.data.put(key.string(), bytes);
+            }
+        );
+    }
+
+    @Override
+    public CompletableFuture<Void> move(final Key source, final Key destination) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Flow.Publisher<ByteBuffer>> value(final Key key) {
+        return CompletableFuture.supplyAsync(
+            () -> {
+                final byte[] bytes = this.data.get(key.string());
+                if (bytes == null) {
+                    throw new IllegalArgumentException(
+                        String.format("No value for key: %s", key.string())
+                    );
+                }
+                return FlowAdapters.toFlowPublisher(Flowable.fromArray(ByteBuffer.wrap(bytes)));
```

Comment:
@olegmoz maybe this line should be divided into smaller ones? For readability reasons.